packages/amplify-e2e-tests/overrides/override-api-gql.ts

```diff
@@ -1,3 +1,15 @@
-export function override(resource: any): any {
+export function override(resource: any, amplifyProjectInfo: any): any {
   resource.api.GraphQLAPI.xrayEnabled = true;
+
+  if (!amplifyProjectInfo || !amplifyProjectInfo.envName || !amplifyProjectInfo.projectName) {
+    throw new Error(`Project info is missing in override: ${JSON.stringify(amplifyProjectInfo)}`);
+  }
+
+  if (amplifyProjectInfo.envName != '##EXPECTED_ENV_NAME') {
+    throw new Error(`Received unexpected envName: ${amplifyProjectInfo.envName}`);
+  }
+
+  if (amplifyProjectInfo.projectName != '##EXPECTED_PROJECT_NAME') {
+    throw new Error(`Received unexpected projectName: ${amplifyProjectInfo.projectName}`);
+  }
```

Comment:
Assertions are injected into override code with placeholders. This is because overrides don't return anything (or even if they do nothing captures return value).
(this pattern is repeated in other tests, with placeholders in e2e tests and hardcoded values in unit tests).